src/values/AbstractValue.js

```diff
@@ -66,6 +66,7 @@ export default class AbstractValue extends Value {
     this.args = args;
     this.hashValue = hashValue;
     this.kind = optionalArgs ? optionalArgs.kind : undefined;
+    this.reactHint = undefined;
```

Comment:
This assignment creates the property for all abstract values, which is not something we want.

Reply:
I'm assigning it to keep the `AbstractValue` monomorphic. If we don't define it, the hidden class of it will change (impacting performance).

Reply:
The performance argument cuts both ways. Generally, I'm loathe to add memory pressure. Having two classes, may be less bad than filling up memory with emptiness. Do you have measurements that suggest this is a clear win?

Reply:
Sorry, I meant this is just something that can have a performance win for V8 (via Node). Otherwise, if we don't define properties that are later added we actually pay more cost in terms of memory and overhead as monomorphic objects can share the same callsites. Here's a good article on this: http://mrale.ph/blog/2015/01/11/whats-up-with-monomorphism.html

Reply:
I agree, we should be measuring these things. One might argue that we should be measuring the performance of **not** ensuring all our classes are monomorphic rather than making everything polymorphic. Especially considering we create enough of them to warrant the hot path of the hidden classes being used, off-setting any memory issues anyway.

I'll revert this change for now because you seem to feel strongly about this and I don't have the time to add in extensive and accurate benchmarking for Prepack right now (although I definitely want to). I'll be in Seattle at the end of the month, I'd be happy to sit down with you and explain some of the ideas I have around benchmarking this sort of heuristic though.